pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -2450,6 +2450,64 @@ The delayed message index bucket time step(in seconds) in per bucket snapshot se
     )
     private long namespaceBundleUnloadingTimeoutMs = 60000;
 
+    @FieldContext(
+            category = CATEGORY_LOAD_BALANCER,
+            doc = "Option to enable the debug mode for the load balancer logics. "
+                    + "The debug mode prints more logs to provide more information "
+                    + "such as load balance states and decisions. "
+                    + "(only used in load balancer extension logics)"
+    )
+    private boolean loadBalancerDebugModeEnabled = false;
```

Comment:
Does this config is necessary? Maybe we can do the same thing via modify log configurations.

Reply:
Some systems do support fine-grained debugging logs for certain features.

I am curious if the pulsar follows this practice, but I think this could be useful to see debug logs from load balancer logic only.

Reply:
Should we mark this configuration as dynamic? Then we can change this configuration in real time.

Reply:
I think we can mark all these configs as dynamic. Updating.

Reply:
The Pulsar broker uses the log4j2, which supports custom log configurations by modifying the log config file `${PULSAR_HOME}/conf/log4j2.yaml`, which supports log control at class level.

Reply:
The Pulsar broker uses the log4j2, which supports custom log configurations by modifying the log config file `${PULSAR_HOME}/conf/log4j2.yaml`, which supports log control at class level.

Reply:
That's another way of fine-configuring the log level, and I assume we can control the log level by the package as well.

But with this dynamic config, we can more easily turn on/off the debugging logs too.

Reply:
OK, but it's a little strange to add a separate param to control debug log information, we also need to add extra judgment logic.